src/components/atoms/InputText/InputText.tsx

```diff
@@ -3,77 +3,66 @@ import styled, { css } from 'styled-components';
 import { colors } from '../../../constants';
 import { getInputTextColor, getBorderColorByStatus } from '../../../helpers/utils';
 import { InputProps } from '../../types';
-import { useThemeContext, AppTheme } from '../../styles/Theme';
+import { useThemeContext, AppThemeProps } from '../../styles/Theme';
 
-type IconWrapperProps = {
-  startIcon?: boolean;
-  /**
-   * The weight of the rendered text.
-   * @default 'regular'
-   */
-  weight?: keyof AppTheme['typography']['weights'];
-  /**
-   * The size you want to render your text at, currently only `13px` | `15px` and `18px` supported.
-   * @default 'body'
-   */
-  size?: keyof AppTheme['typography']['text']['sizes'];
-};
+interface InputThemeProps extends AppThemeProps, InputProps {}
 
 const InputWrapper = styled.div`
   position: relative;
 `;
 
-const IconWrapper = styled.span<IconWrapperProps & { theme: AppTheme }>`
+const IconWrapper = styled.span<InputThemeProps>`
   position: absolute;
   top: 1px;
   bottom: 1px;
   display: flex;
   align-items: center;
   justify-content: center;
-  width: 48px;
-  color: ${({ theme }) => theme.input.iconColor};
-  ${({ startIcon }) =>
-    startIcon
-      ? css`
-          left: 1px;
-          border-top-left-radius: 8px;
-          border-bottom-left-radius: 8px;
-        `
-      : css`
-          right: 1px;
-          border-top-right-radius: 8px;
-          border-bottom-right-radius: 8px;
-        `}
+  width: ${({ theme }: InputThemeProps) => (theme?.input?.iconWidth ? theme.input.iconWidth : '48px')};
+  color: ${({ theme }: InputThemeProps) => theme.input.iconColor};
+${({ startIcon }) =>
+  startIcon
+    ? css`
+        left: 1px;
+        border-top-left-radius: 8px;
+        border-bottom-left-radius: 8px;
+      `
+    : css`
+        right: 1px;
+        border-top-right-radius: 8px;
+        border-bottom-right-radius: 8px;
+      `}
   background-color: ${({ startIcon, theme }) => startIcon && theme.input.iconBackgroundColor};
 `;
 
-const Input = styled.input<InputProps & { theme: AppTheme }>`
+const Input = styled.input<InputThemeProps>`
   width: 100%;
   -webkit-appearance: none;
   outline: none;
-  border-radius: ${({ theme }) => theme.input.borderRadius};
+  border-radius: ${({ theme }: InputThemeProps) => theme.input.borderRadius};
```

Comment:
I think we discussed this before(on the Alert component) regarding the fallback option. best to provide a fallback value for consistency.

Reply:
will add as required prop in zopa theme instead

Reply:
will add as required prop in zopa theme instead